### ISSUE TITLE
api.stopTunnel, disconnect, and kill do not work as expected

### ISSUE THREAD

**Issue by the-vampiire** [bubenshchykov/ngrok#267]
when i try to kill a tunnel and then create a new one (in the same running process) i get an error as it tries to load the config file. 

i traced this down and it is due to the `defaults` function in `utils.js`. a `name` property must be added after the first tunnel is created and it never gets cleared when `disconnect` or even `kill` are called.

```js
function defaults(opts) {
  opts = opts || { proto: "http", addr: 80 };
  // if (opts.name) {
  //   const configPath = opts.configPath || defaultConfigPath();
  //   const config = parse(readFileSync(configPath, "utf8"));
  //   if (config.tunnels && config.tunnels[opts.name]) {
  //     opts = Object.assign(opts, config.tunnels[opts.name]);
  //   }
  // }
  if (typeof opts === "function") opts = { proto: "http", addr: 80 };
  if (typeof opts !== "object") opts = { proto: "http", addr: opts };
  if (!opts.proto) opts.proto = "http";
  if (!opts.addr) opts.addr = opts.port || opts.host || 80;
  if (opts.httpauth) opts.auth = opts.httpauth;
  return opts;
}
```

when i comment out these lines it works as expected. i can call `disconnect()` then `connect()` (with my settings).

i thought maybe i could pass a `configPath` and then set the `name` like below to make simpler options when connecting and also resolve this bug:

```js
const options: Ngrok.Options = {
  name: "service-name",
  configPath: __dirname + "ngrok.yml", // local config for this service
}
```

but the strange thing is `name` is not an option in `Ngrok.Options` so there is no way for me to set this when calling `connect()`. im not sure what purpose the `opts.name` check performs then. 

the options passed to `connect` seem to be passed around and mutated at runtime which leads to this bug. it looks like it is being set in `ngrok/index.js` in `connectRetry` function:

```js
async function connectRetry(opts, retryCount = 0) {
  opts.name = String(opts.name || uuid.v4());
  try {
    const response = await ngrokClient.startTunnel(opts);
    return response.public_url;
  } catch (err) {
    if (!isRetriable(err) || retryCount >= 100) {
      throw err;
    }
    await new Promise((resolve) => setTimeout(resolve, 200));
    return connectRetry(opts, ++retryCount);
  }
}
```

**Comment by philnash** [comment]
Thanks for this investigation! That mutation of the options in `connectRetry` seems to be the issue here. I'll work on something to fix that.

**Comment by the-vampiire** [comment]
cheers man thank you. i was going to suggest exactly what you used (clone deep), i used it in another similar situation!

**Comment by philnash** [comment]
This has been released as version 4.3.1 now.
